portality/tasks/find_discontinued_soon.py

```diff
@@ -0,0 +1,137 @@
+import datetime
+import csv
+import json
+
+from portality.core import app
+from portality.bll import DOAJ
+from portality import models,app_email
+
+from portality.tasks.redis_huey import main_queue
+
+from portality.background import BackgroundTask, BackgroundSummary
+from portality.tasks.helpers import background_helper
+from portality.ui.messages import Messages
+from portality import constants
+
+
+def _date():
+    return (datetime.datetime.today() + datetime.timedelta(days=app.config.get('DISCONTINUED_DATE_DELTA', 1))).strftime(
+        '%Y-%m-%d')
+class DiscontinuedSoonQuery:
+    @classmethod
+    def query(cls):
+        return {
+            "query": {
+                "bool": {
+                    "filter": {
+                        "bool" : {
+                            "must": [
+                                {"term" : {"bibjson.discontinued_date": _date()}},
+                                {"term" : {"admin.in_doaj":True}}
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+# ~~FindDiscontinuedSoonBackgroundTask:Task~~
+
+class FindDiscontinuedSoonBackgroundTask(BackgroundTask):
+    __action__ = "find_discontinued_soon"
+
+    def find_journals_discontinuing_soon(self, job):
+        jdata = []
+
+        for journal in models.Journal.iterate(q=DiscontinuedSoonQuery.query(), keepalive='5m', wrap=True):
+            # ~~->Journal:Model~~
+            jdata.append(journal.id)
+            job.add_audit_message(Messages.DISCONTINUED_JOURNAL_FOUND_LOG.format(id=journal.id))
+
+        return jdata
+
+    def run(self):
+        job = self.background_job
+        journals = self.find_journals_discontinuing_soon(job=job)
+        if len(journals):
+            for j in journals:
+                DOAJ.eventsService().trigger(models.Event(
```

Comment:
Sends one notification per journal. May be slightly awkward if multiple journals discontinue that day for the same editor, but allows to add the action URL to the specific journal.

Reply:
I think this is the appropriate behaviour